exopy_hqc_legacy/instruments/drivers/visa/tektro_awg.py

```diff
@@ -713,7 +843,10 @@ def running(self, value):
         self.clear_output_buffer()
         if value in ('RUN', 1, 'True'):
             self.write('AWGC:RUN:IMM')
-            if self.ask_for_values('AWGC:RST?')[0] not in (1, 2):
+            self.write('AWGC:RST?')
+            time.sleep(delay)
+            values = self.read_values(format=2)
```

Comment:
This is deprecated in pyvisa. It would be best to use a more recent API. Also all `query` method in pyvisa have a `delay` keyword that handle the sleep for you.